### ISSUE TITLE
Fractal store configuration on module level to span all components within module?

### ISSUE THREAD

**Issue by adamlubek** [angular-redux/store#478]
### This is a...
* [x] usage question

### What toolchain are you using for transpilation/bundling?

* [x] angular/cli

Fractal store article implies that it has to be defined on component level (https://github.com/angular-redux/store/blob/master/articles/fractal-store.md). 
My question is: can it be configured on module level so no additional code in each component is required or does it have to be configured for each component?

**Comment by e-schultz** [comment]
Currently it would still need to be configured at the root component of that module.

The `configureSubStore` currently does not accept a parameter for an initial state, although I don't think that would be too hard of a fix to add.

In the meantime though, would something like: 

```ts
export const lazyReducer = (initialState) => (state, action) =>
  action.type === 'ADD_LOC' ?
  { ...state, loc: state.loc + action.payload } :
  state;

// in the component

ngOnInit() {
 this.subStore = this.ngRedux.configureSubStore(
      ['users', userId], lazyReducer(initialState));
}
```
work to solve your use-case?

**Comment by adamlubek** [comment]
thanks for reply but unfortunately your suggestion doesn't solve my use case. I'm using @WithSubStore decorator, here's my setup:

I've got base class from which angular components inherit (this way I don't need to setup substore in each component)

```
@WithSubStore({
    basePathMethodName: 'getBasePath',
    localReducer: rootReducer
  })
export class AppSubStore {
    getBasePath(): PathSelector {
        return ['mySubstore']
    }
}
```


then, rootReducer:

```
export const rootReducer: Reducer<IAppState> = combineReducers({
      something: somethingReducer,
      somethingElse: somethingElseReducer
  }) as Reducer<IAppState>;
```

and example reducer:

```
export interface ISomething {
    value: number;
}

export const INITIAL_SOMETHING_STATE: ISomething = {
    value: 123
}

export function somethingReducer(state: ISomething = INITIAL_SOMETHING_STATE, action: Action): ISomething {
    switch (action.type) {
        case 'CHANGE_VALUE': return { ...state, value: Math.random() };
    }
    return state;
}

```
how can I initialize default state in such setup?

**Comment by flensrocker** [comment]
Why do you use the rootReducer for the substore? I use an additional reducer-function as the localReducer of the substore. If I would add that with combineReducers to the root-store, it will be called twice.

So I have the inital state of the substore at the localReducer like in your example of somethingReducer. And my app-module doesn't have to know anything about the substate and reducers of the sub-module.

**Comment by adamlubek** [comment]
rootReducer is in fact a local reducer defined for lazy loaded code with it's own substore. I see why variable name caused you confusion though, sorry about that.

problem with setup I outlined is that when code containing this implementation is lazy loaded, initial state for substore is initialized only after some action is dispatched. I'd like to initial state to be initialized when code is lazy loaded.

**Comment by flensrocker** [comment]
Ah, I see.

You can use the constructor of the lazy loaded module to dispatch a special init-action. It's a bit ugly but not the worst thing, I guess.

**Comment by adamlubek** [comment]
yeah, that works but I'd prefer to have built in mechanism rather than resorting to work arounds
